tools/github-actions/cascading/src/helpers.ts

```diff
@@ -0,0 +1,143 @@
+import type {BaseLogger} from './cascading';
+
+/**
+ * Sort branches according to semantic versioning
+ *
+ * @param branches
+ */
+export function sortBranches(branches: string[]) {
+  return branches.map(branch => {
+    const extract = branch.match(/release\/([0-9]+)\.([0-9]+)/);
```

Comment:
This should not be based on `cascadingPattern` input?
If my config is `cascadingPattern: publish/*` this function will not work very well (same as the one below) :S

Reply:
We had a discussion on this and we chose to have  a very strict pattern to avoid any issue (unwanted cascading on other repository branches) as mentionned here : 
`This plugin is meant to implement the Bitbucket cascading mechanism on GitHub.
For the moment, the branching model supported is the following one : release/Maj.min[.0-alpha|beta|next|rc] and a default branch from your choice.
This pattern is restrictive on purpose, to prevent any issue where some non-release branches could match as well`
And the cascading Pattern option applies a pre filter only:
`The cascading pattern used as a pre-filter on repositiory branches considered for cascading`.

The idea was that we would create a new feature in case projects want an enhancement on this.

Reply:
As discuss, a compromise would be to:
- Remove the input to not have inconsistency with this code
- Create a task to enhance the plugin with the flexibility bring by the input